frontera/contrib/backends/remote/codecs/json.py

```diff
@@ -91,48 +93,49 @@ def __init__(self, request_model, response_model, *a, **kw):
+                                   method=obj[b'method'],
+                                   headers=obj[b'headers'],
+                                   cookies=obj[b'cookies'],
+                                   meta=obj[b'meta'])
 
     def decode(self, message):
-        message = super(Decoder, self).decode(message)
-        if message['type'] == 'page_crawled':
-            response = self._response_from_object(message['r'])
-            links = [self._request_from_object(link) for link in message['links']]
+        message = dict_to_bytes(super(Decoder, self).decode(message))
+        if message[b'type'] == b'page_crawled':
+            response = self._response_from_object(message[b'r'])
+            links = [self._request_from_object(link) for link in message[b'links']]
             return ('page_crawled', response, links)
-        if message['type'] == 'request_error':
-            request = self._request_from_object(message['r'])
-            return ('request_error', request, message['error'])
-        if message['type'] == 'update_score':
-            return ('update_score', self._request_from_object(message['r']), message['score'], message['schedule'])
-        if message['type'] == 'add_seeds':
+        if message[b'type'] == b'request_error':
+            request = self._request_from_object(message[b'r'])
+            return ('request_error', request, to_native_str(message[b'error']))
+        if message[b'type'] == b'update_score':
+            return ('update_score', self._request_from_object(message[b'r']), message[b'score'], message[b'schedule'])
+        if message[b'type'] == b'add_seeds':
             seeds = []
-            for seed in message['seeds']:
+            for seed in message[b'seeds']:
                 request = self._request_from_object(seed)
                 seeds.append(request)
             return ('add_seeds', seeds)
-        if message['type'] == 'new_job_id':
-            return ('new_job_id', int(message['job_id']))
-        if message['type'] == 'offset':
-            return ('offset', int(message['partition_id']), int(message['offset']))
+        if message[b'type'] == b'new_job_id':
+            return ('new_job_id', int(message[b'job_id']))
+        if message[b'type'] == b'offset':
+            return ('offset', int(message[b'partition_id']), int(message[b'offset']))
         return TypeError('Unknown message type')
 
     def decode_request(self, message):
-        obj = super(Decoder, self).decode(message)
-        return self._request_model(url=obj['url'],
-                                   method=obj['method'],
-                                   headers=obj['headers'],
-                                   cookies=obj['cookies'],
-                                   meta=obj['meta'])
+        obj = dict_to_bytes(super(Decoder, self).decode(message))
+        return self._request_model(url=to_native_str(obj[b'url']),
+                                   method=to_native_str(obj[b'method']),
```

Comment:
method here is converted to native string and above in ``_request_from_object() `` it isn't. I think there is a mistake here.